sdk/python/tests/integration/feature_repos/test_repo_configuration.py

```diff
@@ -221,9 +244,13 @@ def construct_test_environment(
     offline_creator: DataSourceCreator = importer.get_class_from_type(
         module_name, config_class_name, "DataSourceCreator"
     )(project)
+
+    # This needs to be abstracted away for test_e2e_universal which uses a different dataset.
+    df = create_dataset()
     ds = offline_creator.create_data_sources(
-        project, df, field_mapping={"ts_1": "ts", "id": "driver_id"}
+        df, destination=project, field_mapping={"ts_1": "ts", "id": "driver_id"}
```

Comment:
Should this field mapping be hardcoded?

Reply:
Also, why would the destination be set to `project`? How can that be overriden? 

I think the key decision here is still, is `construct_test_environment` === `construct_universal_test_environment` in which case everything should be hardcoded but I as a user should be able to compose my own `custom_construct_test_environment`, or this function should be generalized. I would prefer the former.